### ISSUE TITLE
Menus sometimes appear offset

### ISSUE THREAD

**Issue by mirenradia** [psifidotos/applet-window-appmenu#177]
Hello, I am occasionally finding that menus appear slightly offset relative to the Latte Dock panel on which I have placed this widget as in the screenshot below.
![screenshot_20210917_095824](https://user-images.githubusercontent.com/32646026/133755882-411b9294-309a-4fcb-bc0a-f803321539fe.png)
It is not always like this and sometimes appears as expected (i.e. menus directly below the menu button on the panel) but it would be nice if it always did so.

If it helps, these are the software versions I am using:
* Ubuntu 20.04.3
* KDE Plasma 5.18.7 (on X11)
* KDE Frameworks 5.68.0
* Qt 5.12.8
* Latte Dock 0.9.11
* applet-window-appmenu 0.7.1 (installed following [the installation instructions](https://github.com/psifidotos/applet-window-appmenu/blob/master/INSTALLATION.md))

I'm not sure if it's related to the fact I have two monitors with different resolutions (one 1080p laptop screen with a neighbouring 4K external monitor).

Let me know if there's any further information I can provide in order to debug this.

**Comment by psifidotos** [comment]
I think latest Latte 0.10.x with latest appmenu fixes the situation

**Comment by ericmacedo** [comment]
I can confirm the bug with:

- OS: Manjaro 21.1.4 Pahvo
- KDE: 5.22.5-1
- KDE Frameworks: 5.86.0-1
- Latte Dock 0.10.75
- applet-window-appmenu 0.7.1-2
- Plasma widget style: Kvantum 0.20.1-1

The bug is present both in a single monitor and a dual-head setup.

One thing to notice is that when I navigate through the menu a few times, it kind of "tweaks" the offset until it has the proper offset (no offset at all).

**Comment by mirenradia** [comment]
> One thing to notice is that when I navigate through the menu a few times, it kind of "tweaks" the offset until it has the proper offset (no offset at all).

Yes, this is the behaviour I am also observing.

**Comment by ericmacedo** [comment]
> I can confirm the bug with:
> 
> - OS: Manjaro 21.1.4 Pahvo
> - KDE: 5.22.5-1
> - KDE Frameworks: 5.86.0-1
> - Latte Dock 0.10.75
> - applet-window-appmenu 0.7.1-2
> - Plasma widget style: Kvantum 0.20.1-1
> 
> The bug is present both in a single monitor and a dual-head setup.
> 
> One thing to notice is that when I navigate through the menu a few times, it kind of "tweaks" the offset until it has the proper offset (no offset at all).

I confirmed it with latte 0.10.75

**Comment by psifidotos** [comment]
0.10.75 is too abstract can you confirm it with 0.10.1 or  0.10.2 and provide also a way to reproduce?

**Comment by mirenradia** [comment]
I installed Latte Dock from source using the latest commit on the master branch (commit [`d354cc4`](https://github.com/KDE/latte-dock/tree/d354cc45135d8ec41f96072dad493ccfa530e379)) and am still observing this issue.

**Comment by psifidotos** [comment]
send your latte layout file because I can not reproduce in any case

**Comment by mirenradia** [comment]
Here is my layout file: [My Layout New.layout.latte](https://github.com/psifidotos/applet-window-appmenu/files/7318089/My.Layout.New.layout.latte.txt) (obviously without the .txt extension).

**Comment by mirenradia** [comment]
> * I can not reproduce. In your system do you use any scaling factor for your screen other than 1.0 ?

No, I always use 100% scaling.

> * Your layout is a crashy one, there are too many duplicated ids for applets and containments, if you go to Layouts Editor you will see the errors and warnings in there, I wouldnt trust it if you get crashes with Latte

I suspect this may be due to applets (probably the status bar) that are on panels on displays that are currently not connected. Indeed the layout editor seems to reports the warning for different monitors depending on which ones are connected/connected most recently.

In any case, I have made a simpler layout that is closer to the default one, does not seem to have warnings/errors and reproduces this issue for me:
[Test-Layout.layout.latte](https://github.com/psifidotos/applet-window-appmenu/files/7358290/Test-Layout.layout.latte.txt)

**Comment by a-parhom** [comment]
I'm also having same issue with latest Latte-dock and this applet.

**Comment by TohurTV** [comment]
I only ever get this bug with kvantum based themes and get in on the KDE panel and Latte Panels

**Comment by mirenradia** [comment]
> I only ever get this bug with kvantum based themes and get in on the KDE panel and Latte Panels

This is interesting. I have always used a kvantum based theme but I tried switching to a normal one and the problem also goes away for me.

**Comment by aliquo93** [comment]
I can confirm the same behaviour on latte dock 0.10.8

**Comment by psifidotos** [comment]
does that occur when using Breeze application style and Breeze window decorations?

**Comment by michaldybczak** [comment]
I can confirm this issue and when I had it, I used WhiteSur-dark, with Kvantum (the same theme) and corresponding aurorae theme.

**Comment by murakami-marco** [comment]
I also confirm this behavior. In my case I also use Kvantum themes in X11 session. Currently I'm using version v0.6.0 to avoid the menu's offset. It happens, at least to me, since version v0.7.0.
With the latest version and in Wayland session, it only happens the first time appmenu loads the application menu, If you close and then open again the menu, it appears at the right position

**Comment by simply1233** [comment]
This issues seems to be caused by kvantum menu shadow, because without it there is no offset.

**Comment by virtual-designer** [comment]
Seems like this issue is still not fixed. I'm having the same issue, with Fedora 38 on KDE Plasma 5.27.4.

**Comment by saaivignesh20** [comment]
Yes the issue is not yet fixed. I can confirm with KDE neon 5.27 on KDE Plasma 5.27.8

**Comment by Gpinchon** [comment]
I just encountered this issue with Plasma 6, it was working fine until I applied [this theme](https://github.com/vinceliuice/MacSonoma-kde). Now I have to deactivate the menu shadows...

**Comment by RannyArcher** [comment]
i confirm the same issue but tried to play with the configuration file and edit the following variable (i also removed it)
as i heard from few people here talking about shadows:

`menu_shadow_depth=`


it fixed the offset but i saw that the rounded corners are different.
still rounded but not as leaving the variable with its default value.
but anyway still better than that huge irritating gap.
